src/ui/projects/CreateScenePage.js

```diff
@@ -0,0 +1,366 @@
+    border-radius: 6px;
+  }
+  input {
+    opacity: 0;
+    position: absolute;
+  }
+`;
+
+const RightContent = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex: 1;
+  padding: 30px 30px;
+
+  label[type="button"] {
+    display: flex;
+    margin-bottom: 0;
+    margin-right: 5px;
+  }
+`;
+
+function CreateScenePage({ match, api }) {
+  const [isUploading, setIsUploading] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [sceneId, setSceneId] = useState(null);
+  const [error, setError] = useState(null);
+
+  const [sceneInfo, setSceneInfo] = useState({
+    name: "",
+    creatorAttribution: "",
+    allowRemixing: false,
+    allowPromotion: false
+  });
+
+  const [glbFile, setGlbFile] = useState(null);
+  const [thumbnailFile, setThumbnailFile] = useState(null);
+
+  const [thumbnailUrl, setThumbnailUrl] = useState(null);
+  const [sceneUrl, setSceneUrl] = useState(null);
+
+  const history = useHistory();
+
+  const openScene = useCallback(
+    function() {
+      window.open(sceneUrl);
+    },
+    [sceneUrl]
+  );
+
+  useEffect(
+    function() {
+      async function doInitialLoad() {
+        const sceneId = match.params.sceneId;
+        const isNew = sceneId === "new";
+
+        const scene = isNew ? {} : await api.getScene(sceneId);
+        setSceneInfo({
+          name: scene.name || "",
+          creatorAttribution: (scene.attributions && scene.attributions.creator) || "",
+          allowRemixing: scene.allow_remixing,
+          allowPromotion: scene.allow_promotion
+        });
+        setThumbnailUrl(scene.screenshot_url);
+        setSceneId(scene.scene_id);
+        setSceneUrl(scene.url);
+        setIsLoading(false);
+      }
+      doInitialLoad();
+    },
+    [match, api, setSceneInfo, setThumbnailUrl, setSceneId, setSceneUrl, setIsLoading]
+  );
+
+  const onChangeName = useCallback(
+    function(name) {
+      setSceneInfo({ ...sceneInfo, name });
+    },
+    [sceneInfo, setSceneInfo]
+  );
+
+  const onChangeCreatorAttribution = useCallback(
+    function(creatorAttribution) {
+      setSceneInfo({ ...sceneInfo, creatorAttribution });
+    },
+    [sceneInfo, setSceneInfo]
+  );
+
+  const onChangeAllowRemixing = useCallback(
+    function(allowRemixing) {
+      setSceneInfo({ ...sceneInfo, allowRemixing });
+    },
+    [sceneInfo, setSceneInfo]
+  );
+
+  const onChangeAllowPromotion = useCallback(
+    function(allowPromotion) {
+      setSceneInfo({ ...sceneInfo, allowPromotion });
+    },
+    [sceneInfo, setSceneInfo]
+  );
+
+  const onChangeGlbFile = useCallback(
+    function([file]) {
+      setGlbFile(file);
+    },
+    [setGlbFile]
+  );
+
+  const onChangeThumbnailFile = useCallback(
+    function([file]) {
+      if (thumbnailUrl && thumbnailUrl.indexOf("data:") === 0) {
+        URL.revokeObjectURL(thumbnailUrl);
+      }
+
+      setThumbnailFile(file);
+
+      // For preview
+      const reader = new FileReader();
```

Comment:
I think there's a race condition here should the component unmount before the reader has finished. It's an unlikely condition, but you can probably avoid it by using the FileReader in a `useEffect` hook.

```js
useEffect(() => {
  const reader = new FileReader();
  reader.onload = e => {
    setThumbnailUrl(e.target.result);
  };
  reader.readAsDataURL(thumbnailFile);
  
  return () => {
    reader.abort();
  };
}, [thumbnailFile, setThumbnailUrl])
```